HPCC-16663 Refactoring environment.xml configuration libraries

xwang2713 opened the pull request:
The new library configenv allows create/modify/delete enviornment.xml entries.
This is the first implementation to enhance envgen, a command-line tool to create environment.xml.
A new library: configenv is created by refactoring existing library deployutils to allow re-use the codes to update environment.xml.

This pull request implements the features requested in following JIRA tickets:
HPCC-16663
HPCC-17867
HPCC-15662
HPCC-15636
HPCC-8916
HPCC-8915

A new command-line tool envgen2 is developed. It is back-compatible with most options of envgen. The only option dropped is "-assign" which is replaced with "-add-node".
envgen2 uses above configenv library to create environment.xml and add/modify/remove entries of environment.xml file.

There are lots of new code in this pull request.
There is a design docs which has some description about the code: 
http://10.240.32.242/data3/ming/envgen2/envgen2_refactoring_design.docx

Beside this pull reqest, a wiki discussion page: http://10.240.34.10/wiki/index.php/Refactoring_envgen_Code_Review can also be used to discuss the code review.

Signed-off-by: Xiaoming Wang <xiaoming.wang@lexisnexi.com>

@kenrowland please help to review the code. We can talk about how to proceed this.
@cloLN there are 26 tests for envgen2 which cover all related JIRA and envgen backward compatibility tests. 
@g-pan please help to review envgen2 usage text. I can send you an email with all related text.


<!-- Thank you for submitting a pull request to the HPCC project

 PLEASE READ the following before proceeding.

 This project only accepts pull requests related to open JIRA issues.
 If suggesting a new feature or change, please discuss it in a JIRA issue first.
 If fixing a bug, there should be an issue describing it with steps to reproduce.
 The title line of the pull request (and of each commit within it) should refer to the
 associated issue using the format:

 HPCC-nnnnn Short description of issue

 This will allow the Jira ticket to be automatically updated to refer to this pull request,
 and will ensure that the automatically-generated changelog is properly formatted.
 Where a pull request contains a single commit the pull request title will be set automatically,
 assuming that the commit has followed the proper guidelines.

 Please go over all the following points, and put an `x` in all the boxes that apply. You may find
 it easier to press the 'Create' button first then click on the checkboxes to edit the comment.
-->

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - x ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [x] I have added tests to cover my changes.
  - [x] All new and existing tests passed.
  - [x] I have checked that this change does not introduce memory leaks.
  - [x] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [x] Scalability
  - [x] Performance
  - [x] Security
  - [x] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [x] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Testing:
<!-- Please describe how this change has been tested.-->
http://10.240.32.243/view/custom/job/CE-Platform-JIRA-16663/

<!-- Thank you for taking the time to submit this pull request and to answer all of the above-->

xwang2713 committed:
HPCC-16663 Refactoring environment.xml configuration libraries

The new library configenv is developed by refactoring existing deployutils library to allow re-use
the code to manipulate environment.xml

A new command-line tool envgen2 is developed with above library which allows create/modify/delete
enviornment.xml entries.

This implementation  also covers following JIRA tickets:
  HPCC-17867, HPCC-15662, HPCC-15636, HPCC-8916 and HPCC-8915

Fix return statement for void method.
remove deployment/envgen2/test/.ec2
Add prefix "m_" to class memebers
Ensure to add ftslave instance when add roxie, thor and dropzone nodes
Fixed legacy validating environment.xml warning and memory leak
Fix RoxieFarmProcess setting

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.4.1708 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: 5517ac98f2fd9eaf74d9e4dbc077bf8aa71faae7
Build: success
Install hpccsystems-platform-community_7.0.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |89 |89 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |11 |11 |0 |
|setup (thor) |11 |11 |0 |
|setup (roxie) |11 |11 |0 |
|test (hthor) |790 |790 |0 |
|test (thor) |721 |721 |0 |
|test (roxie) |861 |861 |0 |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|   36 sec (00:00:36) |  173 sec (00:02:53) |   61 sec (00:01:01) |    7 sec (00:00:07) |   28 sec (00:00:28) | 1265 sec (00:21:05) |   19 sec (00:00:19) | 1589 sec (00:26:29) |

kenrowland commented:
@xwang2713 A general comment is that if you are going to use '2' as a new name for the tool, I think it should use the new config2 libs. Otherwise, since it looks to be using the current set of tools, I think it should have a different name. That way, all '2' versions of tools are for the new config manager.

xwang2713 commented:
@kenrowland Ok, I can consider to use "deploytuils2" instead of "configenv". But I'd rather leave "configenv2" for the implementation with your configmgr 2.0 integration. Unless we keep "deployutils" name brand and call them "envgen3" and "deployutils3".

xwang2713 commented:
Update license and test cases.

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.4.1708 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: 338aa942a8b9e2467579d0e99b21d1f4dba3ff6b
Build: success
Install hpccsystems-platform-community_7.0.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |89 |89 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |11 |11 |0 |
|setup (thor) |11 |11 |0 |
|setup (roxie) |11 |11 |0 |
|test (hthor) |790 |790 |0 |
|test (thor) |721 |721 |0 |
|test (roxie) |861 |861 |0 |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|   11 sec (00:00:11) |  170 sec (00:02:50) |   61 sec (00:01:01) |    6 sec (00:00:06) |   27 sec (00:00:27) | 1286 sec (00:21:26) |   19 sec (00:00:19) | 1580 sec (00:26:20) |

xwang2713 commented:
Update envgen2 usage per Greg's comments.
Provide more examples in envgen2 usage.

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.4.1708 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: 54c620b812e53abb05ea71dfcd548c22e94d6116
Build: success
Install hpccsystems-platform-community_7.0.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |89 |89 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |11 |11 |0 |
|setup (thor) |11 |11 |0 |
|setup (roxie) |11 |11 |0 |
|test (hthor) |790 |790 |0 |
|test (thor) |721 |721 |0 |
|test (roxie) |861 |861 |0 |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|    9 sec (00:00:09) |  172 sec (00:02:52) |   61 sec (00:01:01) |    9 sec (00:00:09) |   30 sec (00:00:30) | 1294 sec (00:21:34) |   18 sec (00:00:18) | 1593 sec (00:26:33) |

kenrowland commented on deployment/configenv/xml_jlibpt/EnvHelper.hpp:
```diff
@@ -0,0 +1,121 @@
+
+#define USE_WIZARD 1
+
+enum CONFIG_TYPE { CONFIG_INPUT, CONFIG_ENV, CONFIG_ALL };
```
Why not put the CONFIG_TYPE enum inside the class as a public. The refer to it as EnvConfigOptions::<enum>. That way there is no potential conflict with enum values elsewhere.

kenrowland commented on deployment/configenv/xml_jlibpt/EnvHelper.cpp:
```diff
@@ -0,0 +1,376 @@
+      return XML_TAG_TOPOLOGY;
+   else if (!strcmp(nameLC, "ftslave") || !strcmp(nameLC, "ftslaveprocess"))
+      return "FTSlaveProcess";
+
```
Another option to get the xml tag could be to use a std::map<std::string, std::string>. Always insert the key value as a lowercase string. This would be much faster lookup if speed is an issue.

xwang2713 replied:
The function is to provide more flexible mapping. For example if "fts" is unique and meaningful we may allow to use it for "FTSlaveProcess". I try to simplify input parameter. The real name "FTSlaveProcess" is too long.

kenrowland commented on deployment/configenv/xml_jlibpt/EnvHelper.hpp:
```diff
@@ -0,0 +1,121 @@
+   IPropertyTree * clonePTree(IPropertyTree *src);
+
+   // ip address
+   void processNodeAddress(IPropertyTree *params);
```
Just a general comment that I prefer passing references instead of pointers. The compiler can actually optimize some access that it can't with references. Just my preference.

The exception is when the pointer could be null and the method would check the pointer before dereferencing.

xwang2713 replied:
I am really relatively new to jlib. I basically reference existing code. Jlib implements "Owned" generic class to help cleaning the heap. Still some usage is very subtle and I try to be very careful to change the style. I am still leaning Jlib.

kenrowland commented on deployment/configenv/xml_jlibpt/SWProcess.hpp:
```diff
@@ -0,0 +1,76 @@
+   virtual IPropertyTree * cloneComponent(IPropertyTree *params);
+
+   virtual void computerAdded(IPropertyTree *computerNode, const char* instanceXMLTagName=NULL);
+   virtual void computerUpdated(IPropertyTree *computerNode, const char *oldName, const char *oldIp, const char* instanceXMLTagName=NULL);
```
It's better to start using null_ptr instead of NULL

richardkchapman replied:
Agree - new code should be using null_ptr

kenrowland commented on deployment/configenv/xml_jlibpt/Software.cpp:
```diff
@@ -0,0 +1,295 @@
+{
+   const char *comp = params->queryProp("@component");
+   return getSWComp(comp)->remove(params);
+}
```
return type is void, looks like you are returning the value from the remove call. Should that call change its return value this will break.

kenrowland commented on deployment/configenv/xml_jlibpt/Software.cpp:
```diff
@@ -0,0 +1,295 @@
+   else
+   {
+      out.append(compNameLC);
+   }
```
This is another candidate for a map, as is the next method.

xwang2713 replied:
This method map the various user input name to internal component name. The next method use this to resolve the component name and return component object.

xwang2713 commented on deployment/configenv/xml_jlibpt/ComponentBase.cpp:
```diff
@@ -0,0 +1,384 @@
+
+ComponentBase::ComponentBase(const char* name, EnvHelper * envHelper)
+{
+  this->name.clear().appendf("%s",name);
```
class member "name" is StringBuffer.

richardkchapman replied:
Having a parameter name match a member name is confusing and not really recommended

xwang2713 replied:
Add "m_" to all class members

xwang2713 commented on deployment/configenv/xml_jlibpt/ComponentBase.cpp:
```diff
@@ -0,0 +1,384 @@
+{
+
+   IPropertyTree *pNode = createPTree();
+   return updateNode(pNode, pAttrs);
```
createPTree() will create a Property Tree object. It can't return a NULL unless exception is thrown.

xwang2713 commented on deployment/configenv/xml_jlibpt/ComponentBase.cpp:
```diff
@@ -0,0 +1,384 @@
+{
+  this->name.clear().appendf("%s",name);
+  assert(envHelper);
+  this->envHelper = envHelper;
```
To distinguish member variable with input parameter.

richardkchapman replied:
Our normal convention when passing in a value to be assigned to a member variable (to avoid a name clash) is to add a _ on the front of the parameter name.

xwang2713 commented on deployment/configenv/xml_jlibpt/ComponentBase.hpp:
```diff
@@ -0,0 +1,67 @@
+
+protected:
+   Mutex mutex;
+   EnvHelper * envHelper;
```
EnvHelper is created in ConfigEnv class. It is only deleted in ConfigEnv class destructor.  All other class only get a pointer for reference the EnvHelper object.

xwang2713 commented on deployment/configenv/xml_jlibpt/ConfigEnv.cpp:
```diff
@@ -0,0 +1,318 @@
+{
+   Owned<IEnvironmentFactory> factory = getEnvironmentFactory(false);
+   Owned<IConstEnvironment>  constEnv = factory->loadLocalEnvironment(env);
+   return validateEnv(constEnv);
```
This is for backward compatibility. All of this refactoring process use existing code. There will be an implementation for IConfigEnv interface with config mgr 2.0 code.

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.4.1708 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: 4cbac2c4997603be7c054311eed7e352dbe2e377
Build: success
Install hpccsystems-platform-community_7.0.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |89 |89 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |11 |11 |0 |
|setup (thor) |11 |11 |0 |
|setup (roxie) |11 |11 |0 |
|test (hthor) |804 |804 |0 |
|test (thor) |729 |729 |0 |
|test (roxie) |878 |878 |0 |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|    8 sec (00:00:08) |  174 sec (00:02:54) |   61 sec (00:01:01) |    7 sec (00:00:07) |   28 sec (00:00:28) | 1302 sec (00:21:42) |   19 sec (00:00:19) | 1599 sec (00:26:39) |

kenrowland commented:
@xwang2713 Looks ok to me based on responses

xwang2713 commented:
@richardkchapman this is ready to merge.  It is mainly for internal use like envgen.  There is second phase implementation to use Ken's new config manager 2.0, probably in HPCC 7.4 or 8.0.

richardkchapman commented on deployment/envgen2/test/.ec2/pk-ami-ming-1k.pem:
```diff
@@ -0,0 +1,15 @@
+-----BEGIN RSA PRIVATE KEY-----
```
Is this really supposed to be here??

richardkchapman commented on deployment/envgen2/test/.ec2/cert-ami-ming-1k.pem:
```diff
@@ -0,0 +1,15 @@
+-----BEGIN CERTIFICATE-----
```
Is this supposed to be here?

xwang2713 replied:
I will remove .ec2 directory.

richardkchapman commented on deployment/envgen2/test/src/templates/add_binding_esdl_1.diff:
```diff
@@ -0,0 +1,25 @@
+734a735,744
```
Not convinced this file should be here. Or a lot of other files in this directory, come to that.

xwang2713 replied:
all of test directory is to run test for envgen2. The *.diff files are for checking the results. The test.sh generate *.diff which will be compared with *.diff under templates/.  I can put template environment.xml files. But there will be lots of them and each is about 50-100KB.
test directory will not be packaged. Otherwise not sure where I should put them.

richardkchapman commented:
@wangkx Some style comments and some doubts about whether there are files here that should not be.

xwang2713 commented:
Remove deployment/envgen2/test/.ec2 and squashed.

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.4.1708 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: 96abcc8021d90223988c47fbaac9fd34891dced4
Build: success
Install hpccsystems-platform-community_7.0.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |89 |89 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |11 |11 |0 |
|setup (thor) |11 |11 |0 |
|setup (roxie) |11 |11 |0 |
|test (hthor) |804 |804 |0 |
|test (thor) |729 |729 |0 |
|test (roxie) |878 |878 |0 |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|    9 sec (00:00:09) |  174 sec (00:02:54) |   61 sec (00:01:01) |    7 sec (00:00:07) |   28 sec (00:00:28) | 1479 sec (00:24:39) |   19 sec (00:00:19) | 1777 sec (00:29:37) |

xwang2713 commented:
This pull request owner should be "xwang2713"(Xiaoming Wang) not "wangkx" (Kevin Wang).

wangkx commented:
@xwang2713

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.4.1708 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: 604de2133442af6f9342dc6f9176b98fa5512a93
Build: success
Install hpccsystems-platform-community_7.0.0-rc1.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |89 |89 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |11 |11 |0 |
|setup (thor) |11 |11 |0 |
|setup (roxie) |11 |11 |0 |
|test (hthor) |805 |805 |0 |
|test (thor) |729 |729 |0 |
|test (roxie) |878 |878 |0 |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|   24 sec (00:00:24) |  173 sec (00:02:53) |   94 sec (00:01:34) |   19 sec (00:00:19) |   41 sec (00:00:41) | 1344 sec (00:22:24) |   18 sec (00:00:18) | 1713 sec (00:28:33) |

xwang2713 commented:
@richardkchapman I updated this PR based on comments and tests.  The test directory has some files which are created by diff with original environment.xml. I use these file to compare test results. Otherwise I need save all result environment.xml files.

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.4.1708 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: 9ba8f166b4cc5f8724295c34b7191834ce2c7da3
Build: success
Install hpccsystems-platform-community_7.0.0-rc1.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |94 |94 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |11 |11 |0 |
|setup (thor) |11 |11 |0 |
|setup (roxie) |11 |11 |0 |
|test (hthor) |805 |805 |0 |
|test (thor) |729 |729 |0 |
|test (roxie) |878 |878 |0 |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|   16 sec (00:00:16) |  175 sec (00:02:55) |   95 sec (00:01:35) |   21 sec (00:00:21) |   42 sec (00:00:42) | 1319 sec (00:21:59) |   18 sec (00:00:18) | 1686 sec (00:28:06) |

richardkchapman commented:
@xwang2713 You have not addressed my comments regarding parameter names / use of this ...

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.4.1708 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: bd8a496219504b718f2b1f7a8383485533757fb8
Build: success
Install hpccsystems-platform-community_7.0.0-rc1.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |94 |94 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |11 |11 |0 |
|setup (thor) |11 |11 |0 |
|setup (roxie) |11 |11 |0 |
|test (hthor) |805 |805 |0 |
|test (thor) |729 |729 |0 |
|test (roxie) |878 |878 |0 |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|   19 sec (00:00:19) |  178 sec (00:02:58) |   93 sec (00:01:33) |   19 sec (00:00:19) |   41 sec (00:00:41) | 1327 sec (00:22:07) |   18 sec (00:00:18) | 1695 sec (00:28:15) |

xwang2713 commented:
Add "m_" to all class members.
Create a tar file for all test inputs and results. Delete "src/" directory. During build the tar file will be unpacked to envgen2/test directory.
Fixed some issues found during Docker/Kubernetes testing.

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.4.1708 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: c6067b382e724ca497b2b32b58eab5900cfd4048
Build: success
Install hpccsystems-platform-community_7.0.0-rc1.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |94 |94 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |11 |11 |0 |
|setup (thor) |11 |11 |0 |
|setup (roxie) |11 |11 |0 |
|test (hthor) |805 |805 |0 |
|test (thor) |729 |729 |0 |
|test (roxie) |878 |878 |0 |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|   33 sec (00:00:33) |  175 sec (00:02:55) |   93 sec (00:01:33) |   20 sec (00:00:20) |   42 sec (00:00:42) | 1341 sec (00:22:21) |   19 sec (00:00:19) | 1723 sec (00:28:43) |

richardkchapman commented:
Sorry, I don't like a checked in tar file. Version control can't see changes in it, and I can't easily review its contents

xwang2713 commented:
Remove tar file and restore "src" directory.  Rename directory "templates" to "result_comparison"

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.4.1708 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: 1c12ae6120787eebf3fee30377faa40f56d5644c
Build: success
Install hpccsystems-platform-community_7.0.0-rc1.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |94 |94 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |11 |11 |0 |
|setup (thor) |11 |11 |0 |
|setup (roxie) |11 |11 |0 |
|test (hthor) |805 |805 |0 |
|test (thor) |729 |729 |0 |
|test (roxie) |878 |878 |0 |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|   18 sec (00:00:18) |  180 sec (00:03:00) |   94 sec (00:01:34) |   22 sec (00:00:22) |   44 sec (00:00:44) | 1329 sec (00:22:09) |   20 sec (00:00:20) | 1707 sec (00:28:27) |

richardkchapman commented:
Please squash

xwang2713 commented:
I found one more bug which I will fix soon

xwang2713 commented:
Ensure to add ftslave instance when add roxie, thor and dropzone nodes
Fixed legacy validating environment.xml warning and memory leak
rebased with master
Squashed in the last push

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.4.1708 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: 2903a5d6fc6c984e7de63dab5359b72fea3cdd06
Build: success
Install hpccsystems-platform-community_7.0.0-rc1.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |94 |94 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |11 |11 |0 |
|setup (thor) |11 |11 |0 |
|setup (roxie) |11 |11 |0 |
|test (hthor) |805 |805 |0 |
|test (thor) |729 |729 |0 |
|test (roxie) |878 |878 |0 |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|   22 sec (00:00:22) |  173 sec (00:02:53) |   93 sec (00:01:33) |   19 sec (00:00:19) |   41 sec (00:00:41) | 1348 sec (00:22:28) |   17 sec (00:00:17) | 1713 sec (00:28:33) |

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.4.1708 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: 257893fc76ded86861b0b030200fef9d21a3223a
Build: success
Install hpccsystems-platform-community_7.0.0-rc1.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |94 |94 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |11 |11 |0 |
|setup (thor) |11 |11 |0 |
|setup (roxie) |11 |11 |0 |
|test (hthor) |805 |805 |0 |
|test (thor) |729 |729 |0 |
|test (roxie) |878 |878 |0 |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|   25 sec (00:00:25) |  178 sec (00:02:58) |   95 sec (00:01:35) |   21 sec (00:00:21) |   43 sec (00:00:43) | 1374 sec (00:22:54) |   19 sec (00:00:19) | 1755 sec (00:29:15) |

xwang2713 commented:
Fix RoxieFarmProcess setting for "roxie not license" error in Docker Roxie cluster.

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.4.1708 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: f5581006fe75c6f2525ba5843082e24986483d74
Build: success
Install hpccsystems-platform-community_7.0.0-rc1.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |94 |94 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |11 |11 |0 |
|setup (thor) |11 |11 |0 |
|setup (roxie) |11 |11 |0 |
|test (hthor) |805 |805 |0 |
|test (thor) |729 |729 |0 |
|test (roxie) |878 |878 |0 |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|   21 sec (00:00:21) |  173 sec (00:02:53) |   94 sec (00:01:34) |   19 sec (00:00:19) |   41 sec (00:00:41) | 1355 sec (00:22:35) |   18 sec (00:00:18) | 1721 sec (00:28:41) |
